plugins/module_utils/linode_helper.py

```diff
@@ -283,13 +283,13 @@ def get_all_paginated(
     result = []
     current_page = 1
     page_size = 100
-    num_pages = 1
+    num_pages: Optional[int] = None
 
     if num_results is not None and num_results < page_size:
         # Clamp the page size
         page_size = max(min(num_results, 100), 25)
 
-    while current_page <= num_pages and (
+    while (num_pages is None or current_page <= num_pages) and (
```

Comment:
`num_pages` wasn't being updated here, leading to only the first page being retrieved. This didn't surface until now because generate users are filtering on <= 100 items.